src/clients/python/experimental_api_v2/library/grpcclient.py

```diff
@@ -635,102 +661,94 @@ def wrapped_callback(call_future):
                 error = get_error_grpc(rpc_error)
             callback(result=result, error=error)
 
-        request = self._get_inference_request(inputs, outputs, model_name,
-                                              model_version, request_id,
-                                              parameters)
+        request = _get_inference_request(inputs, model_name, model_version,
+                                         outputs, request_id)
 
         try:
             self._call_future = self._client_stub.ModelInfer.future(request)
             self._call_future.add_done_callback(wrapped_callback)
         except grpc.RpcError as rpc_error:
             raise_error_grpc(rpc_error)
 
-    def _get_inference_request(self, inputs, outputs, model_name, model_version,
-                               request_id, parameters):
-        """Creates and initializes an inference request.
+    def async_sequence_infer(self,
+                             sequence,
+                             inputs,
+                             model_name,
+                             model_version="",
+                             outputs=None,
+                             request_id=None,
+                             end_of_sequence=False):
+        """Run asynchronous sequence inference.
 
         Parameters
         ----------
+        sequence : InferSequence
+            The sequence to be used for the inference request.
         inputs : list
             A list of InferInput objects, each describing data for a input
             tensor required by the model.
-        outputs : list
-            A list of InferOutput objects, each describing how the output
-            data must be returned. Only the output tensors present in the
-            list will be requested from the server.
         model_name: str
             The name of the model to run inference.
         model_version: str
             The version of the model to run inference. The default value
             is an empty string which means then the server will choose
             a version based on the model and internal policy.
+        outputs : list
+            A list of InferOutput objects, each describing how the output
+            data must be returned. If not specified all outputs produced
+            by the model will be returned using default settings.
         request_id: str
             Optional identifier for the request. If specified will be returned
             in the response. Default value is 'None' which means no request_id
             will be used.
-        parameters: dict
-            Optional inference parameters described as key-value pairs.
-
-        Returns
-        -------
-        ModelInferRequest
-            The protobuf message holding the inference request.
-        
+        end_of_sequence: bool
```

Comment:
I'm thinking about how we know when to close the GRPC stream. Does reseting the sequence do that? We should discuss this as well.